span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java

```diff
@@ -79,10 +90,19 @@ public static JaegerSpanNormalizer get(Config config) {
   }
 
   public JaegerSpanNormalizer(Config config) {
-    if (config.hasPath(SpanNormalizerConstants.PII_FIELDS_CONFIG_KEY)) {
-      config.getStringList(SpanNormalizerConstants.PII_FIELDS_CONFIG_KEY).stream()
-          .map(String::toUpperCase)
-          .forEach(tagsToRedact::add);
+    try {
+      if (config.hasPath(SpanNormalizerConstants.PII_KEYS_CONFIG_KEY)) {
+        config.getStringList(SpanNormalizerConstants.PII_KEYS_CONFIG_KEY).stream()
+            .map(String::toUpperCase)
+            .forEach(tagKeysToRedact::add);
+      }
+      if (config.hasPath(SpanNormalizerConstants.PII_REGEX_CONFIG_KEY)) {
+        config.getStringList(SpanNormalizerConstants.PII_REGEX_CONFIG_KEY).stream()
+            .map(Pattern::compile)
+            .forEach(tagRegexPatternToRedact::add);
+      }
+    } catch (Exception e) {
+      LOG.error("An exception occurred while loading redaction configs: ", e);
```

Comment:
Don't swallow this, throw an exception please. Otherwise the application will work without doing any redaction.

Reply:
Discussed it on slack.